### DISCUSSION TITLE
How to dynamically set `run_config` each time a schedule runs?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18281] [Q&A] [answered]
Hello! :wave: I’m not sure what I have is a bug or intended behaviour - I’m trying to configure a schedule with a variable argument to the run config that should be calculated when the schedule launches a run. Something like this:

```my_schedule = ScheduleDefinition(..., run_config={... "time": datetime.now(timezone.utc).strftime("%Y-%m-%d 00:00:00")})```
When I try this locally, everything is fine, with the run config changing dependent on the current date.

However, for the scheduled executions it seems that `datetime.now` is computed on deployment and is then left static; it has supplied the same run config for the last 3 days, all referring to the date of the last deploy before the weekend.

Is this expected? Should I be supplying a variable run config some other way? Or is this indeed a bug that I should raise a ticket for?

Thanks! :pray:

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1701097626457219?thread_ts=1701097626.457219&cid=C01U954MEER).*

**Comment by smackesey** [comment, marked as answer]
You can use `run_config_fn` on `ScheduleDefinition`: https://docs.dagster.io/_apidocs/schedules-sensors#dagster.ScheduleDefinition

Or you can just use a custom eval function that sets `run_config` on `RunRequest`. See the schedule docs: https://docs.dagster.io/concepts/partitions-schedules-sensors/schedules
